ci/checks/check_name_tag_snippets_referenced.py

```diff
@@ -0,0 +1,122 @@
+    for dep in deps:
+        if not shutil.which(dep):
+            raise Exception(f"Must have `{dep}` installed in PATH to run {__file__}")
+
+
+def get_snippet_definitions(target_dir: pathlib.Path) -> List[str]:
+    try:
+        res_snippets = subprocess.run(
+            [
+                "grep",
+                "--recursive",
+                "--binary-files=without-match",
+                "--no-filename",
+                "--ignore-case",
+                "--word-regexp",
+                "--regexp",
+                r"^# <snippet .*name=.*>",
+                str(target_dir),
+            ],
+            text=True,
+            capture_output=True,
+        )
+        res_snippet_names = subprocess.run(
+            ["sed", 's/.*name="//; s/">//; s/version-[0-9\\.]* //'],
+            text=True,
+            input=res_snippets.stdout,
+            capture_output=True,
+        )
+        return res_snippet_names.stdout.splitlines()
+    except subprocess.CalledProcessError as e:
+        raise RuntimeError(
+            f"Command {e.cmd} returned with error (code {e.returncode}): {e.output}"
+        ) from e
+
+
+def get_snippets_used(target_dir: pathlib.Path) -> List[str]:
+    try:
+        res_snippet_usages = subprocess.run(
+            [
+                "grep",
+                "--recursive",
+                "--binary-files=without-match",
+                "--no-filename",
+                "--exclude-dir=versioned_code",
+                "--exclude-dir=versioned_docs",
+                "--ignore-case",
+                "-E",
+                "--regexp",
+                r"```(python|yaml).*name=",
+                str(target_dir),
+            ],
+            text=True,
+            capture_output=True,
+        )
+        res_snippet_used_names = subprocess.run(
+            ["sed", 's/.*="//; s/".*//; s/version-[0-9\\.]* //'],
+            text=True,
+            input=res_snippet_usages.stdout,
+            capture_output=True,
+        )
+        return res_snippet_used_names.stdout.splitlines()
+    except subprocess.CalledProcessError as e:
+        raise RuntimeError(
+            f"Command {e.cmd} returned with error (code {e.returncode}): {e.output}"
+        ) from e
+
+
+def main() -> None:
+    check_dependencies("grep", "sed")
+    project_root = pathlib.Path(__file__).parent.parent.parent
+    docs_dir = project_root / "docs"
+    assert docs_dir.exists()
+    tests_dir = project_root / "tests"
+    assert tests_dir.exists()
+    new_violations = sorted(
+        set(get_snippet_definitions(tests_dir)).difference(
+            set(get_snippets_used(docs_dir))
+        )
+    )
+    if new_violations:
+        print(
+            f"[ERROR] Found {len(new_violations)} snippets which are not used within a doc file."
+        )
+        for line in new_violations:
+            print(line)
+        print(
+            f"[ERROR] Found {len(new_violations)} snippets which are not used within a doc file."
```

Comment:
Non blocking question for @TrangPham and @anthonyburdi -- do we want to use `print()` or `logger()` in these scripts?  Just curious -- thank you.

Reply:
I don't have a strong opinion, but I think if you are running this utility you want the output (vs maybe only at a certain log level). Unless there is another reason to use `logger()`?

Reply:
I don't think we are going to route or capture these elsewhere. Running it locally as a script with print is nice, though logger can also be sent to stdout. I don't see a need for logger here currently.